loader.py

```diff
@@ -0,0 +1,346 @@
+        """
+        boards = []
+        players = []
+        castlings = []
+        en_passants = []
+        half_moves = []
+        full_moves = []
+        fens = []
+        for fen in batch:
+            data = self.preprocess_fen(fen)
+            boards.append(data['board'])
+            players.append(data['player'])
+            castlings.append(data['castling'])
+            en_passants.append(data['en_passant'])
+            half_moves.append(data['half_move'])
+            full_moves.append(data['full_move'])
+            fens.append(data['fen'])
+
+        return {'boards': torch.stack(boards).to(self.device),
+                'players': torch.FloatTensor(players).to(self.device),
+                'castlings': torch.stack(castlings).to(self.device),
+                'en_passants': torch.stack(en_passants).to(self.device),
+                'half_moves': torch.FloatTensor(half_moves).to(self.device),
+                'full_moves': torch.FloatTensor(full_moves).to(self.device)}
+
+    def __iter__(self):
+        r"""
+        Like `Iterator.__iter__()`, but applies preprocessing to yielded data.
+        :return: yields preprocessed batches of samples
+        """
+        for batch in self.iterator:
+            yield self.preprocess_batch(batch)
+
+
+class StandardConvSuite(object):
+    def __init__(self, preprocessor):
+        r"""
+        Preprocesses dictionary of tensors to conv-suitable planes
+        :param preprocessor: info source
+        """
+        self.preprocessor = preprocessor
+
+    @staticmethod
+    def preprocess_batch(batch):
+        """
+        Makes batch suitable for convolutional
+        :param batch:
+        :return:
+        """
+        boards = []
+        for board, player in zip(batch['boards'], batch['players']):
+            if player == 1:
+                boards.append(board[[6, 7, 8, 9, 10, 11, 0, 1, 2, 3, 4, 5]])
+            else:
+                boards.append(board)
+        boards_v = torch.stack(boards)
+        boards_v = torch.cat([boards_v, batch['en_passants']], dim=1)
+        castlings = torch.zeros((boards_v.size()[0], 4, 8, 8)).to(boards_v.device)
+        for idx, (castling, player) in enumerate(zip(batch['castlings'], batch['players'])):
+            if player == 1:
+                for c_idx, component in enumerate(castling[[2, 3, 0, 1]]):
+                    castlings[idx, c_idx, :, :] = component
+            else:
+                for c_idx, component in enumerate(castling):
+                    castlings[idx, c_idx, :, :] = component
+        boards_v = torch.cat([boards_v, castlings], dim=1)
+        return boards_v
+
+    def __iter__(self):
+        r"""
+        Like `Preprocessor.__iter__()`, but applies preprocessing to yielded data.
+        :return: yields preprocessed batches of samples
+        """
+        for batch in self.preprocessor:
+            yield self.preprocess_batch(batch)
+
+
+def getListOfDataset(path, numberOfGenerators=1, keyword=""):

```

Comment:
I would use lowercase letters with underscores rather than lowerCamelCase for function names. And why don't rename parameter `numberOfGenerators` into `n_generators` to be consistent with previous usage.